src/python/pants/backend/python/tasks/resolve_requirements_task_base.py

```diff
@@ -77,11 +77,16 @@ def resolve_requirements(self, interpreter, req_libs):
 
       # We need to ensure that we are resolving for only the current platform if we are
       # including local python dist targets that have native extensions.
-      targets_by_platform = pex_build_util.targets_by_platform(self.context.targets(), self._python_setup)
-      if self._python_native_code_settings.check_build_for_current_platform_only(targets_by_platform):
+      # TODO: We need to check all of the targets in the build graph to determine if any of them
+      # contain native code. Making the generated python_requirement_library() from
+      # BuildLocalPythonDistributions depend on the C/C++ sources that it builds might make this
+      # cleaner.
+      all_targets = self.get_targets()
+      if self._python_native_code_settings.check_build_for_current_platform_only(all_targets):
```

Comment:
I'm slightly concerned that in the case that some things being built have native deps, and some don't, we'll only build for the current platform for all of those targets.

But I guess this is probably an ok trade-off until such time as we do per-root resolves. We need to pick one way or the other, so...